docs/sdg/knowledge-doc-ingestion.md

```diff
@@ -0,0 +1,96 @@
+
+- **Example Workflow**:
+  ```bash
+  ilab document format --input path/to/document.pdf --output path/to/schema
+  ```
+  This command will ingest a document, process it into the instructlab schema, and output the result for further use in SDG or RAG workflows.
+
+### 3.4 Simplifying Git-Based Workflows for Users
+
+- **Current Challenge**: Knowledge documents are stored in Git-based repositories, which may be unfamiliar to many users.
+- **Proposed Solution**:
+  - Allow users to input a local directory and provide an automated script that:
+    1. Initializes a Git repository.
+    2. Creates the necessary branches.
+    3. Organizes files into the required structure.
+  
+  By abstracting the Git setup process, we can retain Git’s benefits (version control, backups) while simplifying the interface for non-technical users.
+
+- **Implementation Example**:
+  ```bash
+  ./setup_git_repo.sh --input /path/to/local/docs
+  ```
+  This script automates the process of structuring knowledge documents for ingestion.
+
+### 3.5 Workflow Visualization
+
+Here is a conceptual diagram illustrating the workflow from document ingestion to schema conversion and chunking:
+
+![Knowledge_Document_Ingestion_Workflow](https://github.com/user-attachments/assets/06504b1b-bc8f-4909-b6a2-732a056613c5)
+
+The pipeline begins with the ingestion of knowledge document, passes through a conversion step to instructlab schema using **Docling**, and then processes the document into a format usable by SDG and RAG workflows.
+
+## 4. Future Enhancements
+
+### 4.1 Support for Additional Data Types
+
+To extend beyond text-based documents, the pipeline will explore handling other formats, including:
+- **Audio and Video**: Incorporating media formats will require modifications to the schema and additional processing capabilities.
+- **Visual Language Models (VLMs)**: We will collaborate with the research team to align this work with visual data processing tools and extend the pipeline to handle multimedia.
+
+### 4.2 Refined Chunker Library
+
+- **Standalone Library**: The document chunking functionality will eventually be refactored into a dedicated chunking library or utility within the SDG module. This separation will make it easier to maintain and extend in future iterations.
+  
+- **Performance Optimizations**: Ongoing work will aim to reduce the time and resources needed for large-scale document chunking, particularly for multi-format documents like PDFs containing both text and images.
+
+## 5. InstructLab Schema Overview
+
+### Key Components:
+- **Docling JSON Output**: The output from Docling will be the instructlab schema, which serves as the backbone for both SDG and RAG workflows. For specific details around the leaf node path or timestamp, we will include that as a part of the file nomenclature.
```

Comment:
As part of the ingestion command, should we consider a flag where the pipeline could augment the metadata of the final output like `--metadata ./path-to-metadata.json` to add information such as attribution, timestamps, ilab version, schema version, etc.?